server/database/quieres/filterEstates.js

```diff
@@ -0,0 +1,38 @@
+const connection = require('../config/connection');
+
+module.exports = ({
+  type, category, location, price, roomNumbers, bathRooms, space,
+}) => {
+  let text = 'SELECT * FROM estates WHERE ';
+  const values = [];
+
+  if (type) {
+    text += `type = $${values.length + 1} AND `;
```

Comment:
Never in your life use [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) for SQL queries.

Reply:
Also, one piece of advice for everyone @Mu7ammadAbed @muhammadharoun @ElhamFadel @mohammedsalah7, especially those who intend to specialize in Backend, you *MUST* at least once check a security course.

Reply:
Can you suggest us one? @NoureldeanSaed